strawberryfields/ops.py

```diff
@@ -1615,46 +1645,123 @@ def __or__(self, reg):
 class Interferometer(Decomposition):
     r"""Apply a linear interferometer to the specified qumodes.
 
-    This operation uses the :ref:`Clements decomposition <clements>` to decompose
+    This operation uses either the :ref:`rectangular decomposition <rectangular>`
+    or triangular decomposition to decompose
     a linear interferometer into a sequence of beamsplitters and
     rotation gates.
 
+    By specifying the keyword argument ``mesh``, the scheme used to implement the interferometer
+    may be adjusted:
+
+    * ``mesh='rectangular'`` (default): uses the scheme described in
+      :cite:`clements2016`, resulting in a *rectangular* array of
+      :math:`M(M-1)/2` beamsplitters:
+
+      .. figure:: ../_static/clements.png
+          :align: center
+          :width: 30%
+          :target: javascript:void(0);
+
+      Local phase shifts appear in the middle of the beamsplitter array.
+      Use ``mesh='rectangular_phase_end`` to instead commute all local phase shifts
+      to the end of the beamsplitter array.
+
+      By default, the interferometers are decomposed into :class:`~.BSgate` operations.
+      To instead decompose the interferometer using the :class:`~.ops.MZgate`,
+      use ``mesh='rectangular_symmetric'``.
+
+    * ``mesh='triangular'``: uses the scheme described in :cite:`reck1994`,
+      resulting in a *triangular* array of :math:`M(M-1)/2` beamsplitters:
+
+      .. figure:: ../_static/reck.png
+          :align: center
+          :width: 30%
+          :target: javascript:void(0);
+
+      Local phase shifts appear at the end of the beamsplitter array.
+
     Args:
         U (array[complex]): an :math:`N\times N` unitary matrix
-        tol (float): the tolerance used when checking if the matrix is unitary:
-            :math:`|UU^\dagger-I| \leq` tol
+        mesh (str): the scheme used to implement the interferometer.
+            Options include:
+
+            - ``'rectangular'`` - rectangular mesh, with local phase shifts
+              applied between interferometers
+
+            - ``'rectangular_phase_end'`` - rectangular mesh, with local phase shifts
+              placed after all interferometers
+
+            - ``'rectangular_symmetric'`` - rectangular mesh, with local phase shifts
+              placed after all interferometers, and all beamsplitters decomposed into
+              symmetric beamsplitters
```

Comment:
```suggestion
              pairs of symmetric beamsplitters and phase shifters
```

Reply:
I did not look at all lines of code in detail. If you would like me to do that, let me know and I will do that.